### PR TITLE
Sc 1511

### PR DESCRIPTION
**Pull request by amit-tarento**
Changes for feed api.

### CODE REVIEW

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- service/app/controllers/feed/validator/FeedRequestValidator.java  2
- service/test/controllers/feed/validator/FeedRequestValidatorTest.java  1
- service/test/controllers/feed/FeedControllerTest.java  3
         

Complexity decreasing per file
==============================
+ service/app/controllers/feed/FeedController.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-lms-service/sunbird-lms-service/pullRequest?prid=4560026&bid=15335065)

File: service/app/controllers/feed/validator/FeedRequestValidator.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package controllers.feed.validator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.exception.ProjectCommonException;
+
+/** This call will validate the Feed API request */
+public class FeedRequestValidator {
+  public static void userIdValidation(String accessTokenUserId, String requestUserId) {
+    if (!StringUtils.endsWithIgnoreCase(accessTokenUserId, requestUserId)) {
```
Comment (manzarul) [inline service/app/controllers/feed/validator/FeedRequestValidator.java]: why u want to match prefix with  equal ignore case?
if both are null then it will be true that condition need to be handle

File: service/test/controllers/feed/FeedControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  @Test
+  public void testGetUserFeed() {
+    Result result = performTest("/v1/user/feed/1234567890", "GET");
+    assertEquals(getResponseCode(result), ResponseCode.success.getErrorCode().toLowerCase());
+    assertTrue(getResponseStatus(result) == 200);
+  }
+
+  public Result performTest(String url, String method) {
```
Comment (manzarul) [inline service/test/controllers/feed/FeedControllerTest.java]: performTest, getResponseCode,getResponseStatus can we put it as re-usable.

File: service/test/controllers/feed/validator/FeedRequestValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.junit.Test;
+import org.sunbird.common.exception.ProjectCommonException;
+
+public class FeedRequestValidatorTest {
+
+  @Test
+  public void userIdValidationTestSuccess() {
+    Assert.assertTrue(FeedRequestValidator.userIdValidation("123-456-789", "123-456-789"));
+  }
+
```
Comment (manzarul) [inline service/test/controllers/feed/validator/FeedRequestValidatorTest.java]: add some test case of prefix match and when both are null or any one is null.

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- service/test/controllers/feed/FeedControllerTest.java  1
- service/test/controllers/feed/validator/FeedRequestValidatorTest.java  1
- service/app/controllers/feed/validator/FeedRequestValidator.java  2
- service/test/controllers/BaseApplicationTest.java  1
         

Complexity decreasing per file
==============================
+ service/app/controllers/feed/FeedController.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-lms-service/sunbird-lms-service/pullRequest?prid=4560026&bid=15335065)

**Comment by rjshrjndrn** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- service/app/controllers/feed/validator/FeedRequestValidator.java  2
- service/test/controllers/feed/FeedControllerTest.java  1
- service/test/controllers/feed/validator/FeedRequestValidatorTest.java  1
- service/test/controllers/BaseApplicationTest.java  1
         

Complexity decreasing per file
==============================
+ service/app/controllers/feed/FeedController.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/sunbird-lms-service/sunbird-lms-service/pullRequest?prid=4560026&bid=15335065)
